qiskit_nature/second_q/operators/sparse_label_op.py

```diff
@@ -166,9 +190,7 @@ def _add(self, other: SparseLabelOp, qargs: None = None) -> SparseLabelOp:
         other_unique = {key: other._data[key] for key in other._data.keys() - self._data.keys()}
         new_data.update(other_unique)
 
-        register_length = max(self.register_length, other.register_length)
-
-        return self.__class__(new_data, register_length, copy=False)
+        return self._new_instance(new_data, other=other)
```

Comment:
I cannot comment on this above directly 
```
        if not isinstance(other, SparseLabelOp):
            raise ValueError(
                f"Unsupported operand type(s) for +: 'SparseLabelOp' and '{type(other).__name__}'"
            )
```
But this check would/should depend on sub-class type which may  only be able to add like classes. Should we change this instance test as well as the hard coded SparseLabelOp in the message, since I I try to add something not supported for FermionicOp it would be nice to see that name in the message not some parent class name


Also a comment on equiv and eq below since I cannot comment there directly either. They check other and if not a SparseLableOp they will return False. So its not a check involving subclasses eg FermionicOp to FermionicOp which will pass fine though, but so would FermionicOp and SpinOp in this case. I imagine the latter would most likely always fail anyway since `data` would never be the same - even if in the unlikely case the matrices were the same values the keys would be different. If we are adjusting the add instance checking though maybe we want to consider it in these instances too.

Reply:
Yep thanks for noticing :+1: